### ISSUE TITLE
#!/bin/bash shebang doesn't work on systems where bash isn't in the bin folder

### ISSUE THREAD

**Issue by john-parton** [pia-foss/manual-connections#105]
I'm on FreeBSD 12.

Installing bash with `pkg install bash` puts the bash binary at `/usr/local/bin/bash`.

Solution is to replace the shebang with a call to `env` to get the correct path for bash.

Incorrect

```bash
#!/bin/bash
```

Correct:

```bash
#!/usr/bin/env bash
```

**Comment by john-parton** [comment]
Additionally, we may want to update the README to note that `bash` is a requirement.

**Comment by g00nix** [comment]
Thank you for the PR.

Did you have time to test that this works on all the distributions listed in the README?
- Arch
- Artix
- Fedora 32, 33
- Manjaro
- PureOS amber
- Raspberry Pi OS 2020-08-20
- Ubuntu 18.04, 20.04

**Comment by john-parton** [comment]
No, and I don't intend to install 8 operating system to do so.

Using `/usr/bin/env` for the shebang is more-or-less the standard way to make a script cross platform.

Feel free to close my PR.

**Comment by milahu** [comment]
> test that this works on all the distributions listed in the README?

`/usr/bin/env` is a de-facto stanard in the linux/unix world,
kind-of like the task manager combo Ctrl+Alt+Del in the windows world

so on the opposite, it will be hard to find a distro without `/usr/bin/env`

> there's only /bin/env at least on SCO OpenServer 5.0.6 and Cray Unicos 9.0.2 (although the latter is only of historical interest). On some other Linux distributions (Redhat) it's located in /bin and /usr/bin/ contains a symbolic link pointing to it.

https://unix.stackexchange.com/questions/111802/is-there-any-guarantee-that-usr-bin-env-will-exist

<details>

> Env is part of **coreutils** but the location depends on the distro you are using.

> > On both LFS and Debian env is found in /usr/bin, RHEL (6) has env in /bin.
>
> In RHEL-6 there should be a symlink: /usr/bin/env -> ../../bin/env

> The long standing recommendation for scripting is to begin with the line:
> ```
> #!/usr/bin/env xxxx
> ```

https://www.linuxquestions.org/questions/linux-software-2/usr-bin-env-does-not-exist-4175476804/

> Some info from various *nix systems:
> 
> OS | env path
> -- | --
> OpenBSD | /usr/bin/env
> FreeBSD | /usr/bin/env
> Debian | /usr/bin/env
> Ubuntu | /usr/bin/env
> CentOS | /usr/bin/env
> macOS | /usr/bin/env
> SUSE | /usr/bin/env
> RHEL | /usr/bin/env
> NetBSD | /usr/bin/env
> Solaris | /usr/bin/env

https://www.cyberciti.biz/tips/finding-bash-perl-python-portably-using-env.html

openwrt

```
root@OpenWrt:~# which env
/usr/bin/env
```

archlinux

> usr/bin/env

https://archlinux.org/packages/core/x86_64/coreutils/ # View the file list for coreutils

nixos

```
$ which env
/run/current-system/sw/bin/env

$ file -L -i /usr/bin/env
/usr/bin/env: application/x-executable; charset=binary
```

</details>

[does this answer your question??](https://www.youtube.com/watch?v=Zr0xwgqw-BE#title=Vindaloop%20Customer%20Service%20South%20Park) ; )

**Comment by g00nix** [comment]
> does this answer your question?? ; )

Actually yes. I will push to master in the near future. There are also other modifications that have to be pushed.

**Comment by g00nix** [comment]
Sorry for closing the issue. That was a miss-click. Also thank you for your patience on the topic.

**Comment by g00nix** [comment]
I am going through this ticket now.

> No, and I don't intend to install 8 operating system to do so.

I don't think this is the correct attitude. You are pushing changes that other people have to test. It takes a few seconds to sed the shebang in all files, but it takes a lot more time to test if these changes do not break anything.

We did however install FreeBSD and we did test the modification. Indeed, the scripts now start, however they crashes at later stages. So even with these modifications, the scripts do not fully work on FreeBSD.

> Using `/usr/bin/env` for the shebang is more-or-less the standard way to make a script cross platform.

It is, but all changes must be tested before pushing them to the master branch, no matter how trivial the modifications are.

> Feel free to close my PR.

I rebased the commit into the testing branch. I will create an automated test environment so that we spend less time validating changes like this.
